src/server/components/lists/find/handlers/translators-interpreters/services.ts

```diff
@@ -0,0 +1,54 @@
+import type { Request, Response } from "express";
+import { sanitiseServices } from "server/components/lists/find/helpers/sanitiseServices";
+
+export function get(req: Request, res: Response) {
+  res.render("lists/find/translators-interpreters/services.njk", { values: req.session.answers?.services ?? [] });
+}
+
+export function post(req: Request, res: Response) {
+  const { services } = req.body;
+
+  const sanitisedServices = sanitiseServices(services);
+  if (sanitisedServices.length === 0) {
+    req.flash("error", "Select the services you need");
+    res.redirect("services");
+    return;
+  }
+
+  req.session.answers!.services = sanitisedServices;
+
+  const requiresTranslation = sanitisedServices.includes("translation");
+  const requiresInterpretation = sanitisedServices.includes("interpretation");
+
+  if (!requiresTranslation) {
+    delete req.session.answers?.translationTypes;
+  }
+
+  if (!requiresInterpretation) {
+    delete req.session.answers?.interpretationTypes;
+  }
+
+  const { interpretationTypes = [], translationTypes = [] } = req.session.answers ?? {};
+
+  let shouldRedirectToTypes = false;
+
+  if (requiresTranslation && translationTypes.length === 0) {
+    shouldRedirectToTypes = true;
+  }
+
+  if (requiresInterpretation && interpretationTypes.length === 0) {
+    shouldRedirectToTypes = true;
+  }
+
+  if (req.query.return === "results") {
+    if (shouldRedirectToTypes) {
+      res.redirect("types?return=results");
```

Comment:
what does return=results do here?

Reply:
returns the user back to results page if detected in the query

Reply:
yep, I now get the intent behind it after reviewing the handlers.